src/pages/home/search/SearchModal.tsx

```diff
@@ -32,12 +50,38 @@ function SearchModal() {
     }
   };
 
-  const onSearch = (event: React.ChangeEvent<HTMLInputElement>) => {
-    const recentSearch = [...savedSearch];
-    // currently it is saving each word is typed, need to handle with debounce
-    recentSearch.push(event.target.value);
-    setSavedSearch(recentSearch);
+  const onInputValueChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setInputValue(event.target.value);
+    setCurrHoveredOption(undefined);
+  };
+
+  const onSearch = (newSearch: string, doNavigate = true) => {
+    setSavedSearch((prev) => {
+      const updatedSearch = [...prev, newSearch];
+      const uniqueSearch = [...new Set(updatedSearch)];
+      return uniqueSearch;
+    });
+    setIsOpen(false);
+    setInputValue("");
+    if (doNavigate) navigate(`/${newSearch}`);
```

Comment:
In autoComplete search, when the user is clicking on the searched list, it's getting added to the previous list params i.e./tomato/soup. which results in a 404 error page

![image](https://github.com/chingu-voyages/v46-tier2-team-14/assets/8060579/803930e6-2946-4f32-a131-0b3f75dbc074)